aea/configurations/base.py

```diff
@@ -595,13 +597,13 @@ def __init__(
         build_entrypoint: Optional[str] = None,
         build_directory: Optional[str] = None,
         class_name: str = "",
-        protocols: Optional[Set[PublicId]] = None,
-        connections: Optional[Set[PublicId]] = None,
-        restricted_to_protocols: Optional[Set[PublicId]] = None,
-        excluded_protocols: Optional[Set[PublicId]] = None,
+        protocols: Optional[Set[ExtendedPublicId]] = None,
```

Comment:
You need to accept the union of both PublicId and ExtendedPublicID for backwards compatibility. Or create a type which expresses that.